pykafka/producer.py

```diff
@@ -140,6 +141,14 @@ def __init__(self,
         self._update_lock = self._cluster.handler.Lock()
         self.start()
 
+        def cleanup(obj):
+            if obj._running:
+                obj.stop()
```

Comment:
Since this function is nearly identical to `self.__exit__`, it seems that we could solve this issue by simply using `with` in the client code. Am I missing something?